web/src/lib/data-hooks/useAddressErrors.tsx

```diff
@@ -0,0 +1,57 @@
+import { getErrorStateForAddressField } from "@/components/profile/getErrorStateForAddressField";
+import { AddressContext } from "@/contexts/addressContext";
+import { AddressFields, FieldsForAddressErrorHandling } from "@/lib/types/types";
+import { useContext } from "react";
+
+type AddressErrorsResponse = {
+  doesFieldHaveError: (field: FieldsForAddressErrorHandling) => boolean;
+  doSomeFieldsHaveError: (fields: AddressFields[]) => boolean;
+  getFieldErrorLabel: (field: AddressFields) => string;
+};
+
+export const useAddressErrors = (): AddressErrorsResponse => {
```

Comment:
More of a future state refactor note, it could be cool to have a more generic useFieldErrors hook that takes in the context and validated fields as props.

Reply:
added ticket